hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemMkDir.java

```diff
@@ -197,6 +202,25 @@ public void testGetPathPropertyCalledOnMkdir() throws Exception {
 
   }
 
+  @Test
+  public void testGetPathPropertyCalledOnMkdirFallbackDFS() throws Exception {
+    Assume.assumeTrue(getFileSystem().getAbfsStore().getPrefixMode() == PrefixMode.BLOB);
+    AzureBlobFileSystem fs = Mockito.spy(getFileSystem());
+    AzureBlobFileSystemStore store = Mockito.spy(fs.getAbfsStore());
+    AbfsConfiguration abfsConfig = Mockito.spy(store.getAbfsConfiguration());
+    Mockito.doReturn(true).when(abfsConfig).shouldMkdirFallbackToDfs();
```

Comment:
Since, not all would use config.shouldRedirectToDfsForMkdirs, lets not spy the the abfsConfig.
Instead as discussed in previous comment, lets create clone of abfsConfig present, create fs out of it and then proceed with the test.